inference-engine/src/transformations/src/transformations/common_optimizations/dropout_with_random_uniform_replacer.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/dropout_with_random_uniform_replacer.hpp"
+
+#include <memory>
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+
+#include "itt.hpp"
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::DropoutWithRandomUniformReplacer, "DropoutWithRandomUniformReplacer", 0);
+
+ngraph::pass::DropoutWithRandomUniformReplacer::DropoutWithRandomUniformReplacer() {
+    MATCHER_SCOPE(DropoutWithRandomUniformReplacer);
+    const auto data_pattern = ngraph::pattern::any_input();
+    const auto shape_of_pattern = ngraph::pattern::wrap_type<opset8::ShapeOf>({data_pattern});
+    const auto ru_min_const_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+    const auto ru_max_const_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+    const auto random_uniform_pattern = ngraph::pattern::wrap_type<opset8::RandomUniform>(
+        {shape_of_pattern, ru_min_const_pattern, ru_max_const_pattern},
+        pattern::consumers_count(1));
+    const auto add_const_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+    const auto add_pattern = ngraph::pattern::wrap_type<opset8::Add>({random_uniform_pattern, add_const_pattern});
+
+    const auto floor_pattern = ngraph::pattern::wrap_type<opset8::Floor>({add_pattern});
+
+    const auto mul_input_pattern = ngraph::pattern::any_input();
+    const auto mul_pattern = ngraph::pattern::wrap_type<opset8::Multiply>({floor_pattern, mul_input_pattern});
+
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto pattern_map = m.get_pattern_value_map();
```

Comment:
Please avoid excess copies.
```suggestion
        const auto & pattern_map = m.get_pattern_value_map();
```

Reply:
We can't make pattern_map const as we use operator[] to access its elements and this operator is not const.

Reply:
You shouldn't use [] to access alements in map. Please use at() instead and make this map constant.

Reply:
Done.